### DISCUSSION TITLE
Is there a way to automatically determine the "d" suffix on Windows debug builds for *some* packages?

### DISCUSSION THREAD

**Discussion by GasimGasimzada** [premake/premake-core#2138] [Q&A] [answered]
I am trying to migrate to using vcpkg to install dependencies for my project and then link them using premake. I have came across a problem, for debug builds, some packages add `d` suffix while others do not. So, I wanted to know if there is a convenient way that I can do something like this:

```lua
links { "imgui" }
```

And premake automatically checks both `imgui.lib` and `imguid.lib`. Otherwise, I need to constantly need to do something like this:

```lua
filter { "system:windows", "configurations:Debug" }
        links { "imgui" }
    
filter { "system:windows", "configurations:Release" }
       links { "imguid" }
```

**Comment by nickclark2016** [comment, marked as answer]
To my awareness, this is the only way to do it. That said, you could write a lua function to handle it for you.

**Reply by GasimGasimzada** [reply]
Is there a method to check if a library exists in libdirs? I tried to use `os.findlib` but function always returns nil for me.

**Reply by nickclark2016** [reply]
You need to specify the paths to look in as the second parameter. If not, it just looks in the system include directories.

**Reply by GasimGasimzada** [reply]
I ended up keeping it simple and creating a method that automatically appends `d` suffix for debug configuration:

```lua
function linkSuffixed(...)
    function appendDebug(arr)
        local t = {}
        for k, v in pairs(arr) do
            t[k] = v .. 'd'
        end
    
        return t
    end

    filter { "configurations:Debug or configurations:Test" }
        links(appendDebug(...))

    filter { "configurations:Release or configurations:Profile" }
        links { ... }

    filter{}
end

-- Usage
linkSuffixed { "imgui", "mikktspace" }
```
